lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -1453,4 +1455,155 @@ class TTIRLoadSystemDesc
   }
 };
 
+class GatherOpRewritePattern : public OpRewritePattern<GatherOp> {
+public:
+  using OpRewritePattern<GatherOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(GatherOp op,
+                                PatternRewriter &rewriter) const final {
+    // llvm::outs() << "Running matchAndRewrite" << "\n";
+    // bool reduceWeightTensorDim = false;
+    auto outputType = mlir::cast<RankedTensorType>(op.getResult().getType());
+    auto shape = outputType.getShape();
+    auto hiddenDim = shape[1];
```

Comment:
Is hidden dim guaranteed to be dim 1?  Negative indexing is probably safer, i.e. `shape[shape.size() - 1]` or whatever dim this is referring to.